kratos/mpi/includes/mpi_communicator.h

```diff
@@ -74,673 +67,978 @@ namespace Kratos
-    typedef MeshType::NodesContainerType NodesContainerType;
-
-    /** Iterator over the nodes. This iterator is an indirect
-        iterator over Node::Pointer which turn back a reference to
-        node by * operator and not a pointer for more convenient
-        usage. */
-    typedef MeshType::NodeIterator NodeIterator;
-
-    /** Const iterator over the nodes. This iterator is an indirect
-        iterator over Node::Pointer which turn back a reference to
-        node by * operator and not a pointer for more convenient
-        usage. */
-    typedef MeshType::NodeConstantIterator NodeConstantIterator;
-
-    /** Iterator over the properties. This iterator is an indirect
-        iterator over Properties::Pointer which turn back a reference to
-        properties by * operator and not a pointer for more convenient
-        usage. */
-
-    /// Properties container. Which is a vector set of Properties with their Id's as key.
-    typedef MeshType::PropertiesContainerType PropertiesContainerType;
-
-    /** Iterator over the Properties. This iterator is an indirect
-        iterator over Node::Pointer which turn back a reference to
-        node by * operator and not a pointer for more convenient
-        usage. */
-    typedef MeshType::PropertiesIterator PropertiesIterator;
-
-    /** Const iterator over the Properties. This iterator is an indirect
-        iterator over Properties::Pointer which turn back a reference to
-        Properties by * operator and not a pointer for more convenient
-        usage. */
-    typedef MeshType::PropertiesConstantIterator PropertiesConstantIterator;
 
-    /** Iterator over the properties. This iterator is an indirect
-        iterator over Properties::Pointer which turn back a reference to
-        properties by * operator and not a pointer for more convenient
-        usage. */
-
-    /// Element container. A vector set of Elements with their Id's as key.
-    typedef MeshType::ElementsContainerType ElementsContainerType;
-
-    /** Iterator over the Elements. This iterator is an indirect
-        iterator over Elements::Pointer which turn back a reference to
-        Element by * operator and not a pointer for more convenient
-        usage. */
-    typedef MeshType::ElementIterator ElementIterator;
-
-    /** Const iterator over the Elements. This iterator is an indirect
-        iterator over Elements::Pointer which turn back a reference to
-        Element by * operator and not a pointer for more convenient
-        usage. */
-    typedef MeshType::ElementConstantIterator ElementConstantIterator;
+template<class TValue> struct SendTools;
 
-    /// Condintions container. A vector set of Conditions with their Id's as key.
-    typedef MeshType::ConditionsContainerType ConditionsContainerType;
+template<> struct SendTools<int>
+{
+    typedef int SendType;
+    constexpr static bool IsFixedSize = true;
+    constexpr static std::size_t MessageSize = 1;
```

Comment:
What is the block size? A comment would be appreciated. Or another constexpr.

Reply:
I believe it is the `MessageSize`? It is intended to be the size of a single send, measured in `SendType` (formally, `sizeof(ValueType)/sizeof(SendType)`). If so, I can just rename it to be clearer, since I do not particularly like the current name anyway....